### ISSUE TITLE
Re-add support for amazonlinux:1, centos:7, and ubuntu:14.04

### ISSUE THREAD

**Issue by jszwedko** [vectordotdev/vector#13183]
The newer cross images build on Ubuntu 16.04 which ends up linking against a newer libc and libc++ than is available on ubuntu:14.04, centos:7, amazonlinux:1. We dropped the release tests for them in https://github.com/vectordotdev/vector/pull/13146 to unblock the nightly build, but we should explore re-adding them by linking against older versions of libc/libc++ since these OS's are not yet EOL.

**Comment by darjeeling** [comment]
IMHO, there is still many centos 7 servers alive.

**Comment by leandrojmp** [comment]
Is this targetting any release at the moment? 

I was looking if this [another issue](https://github.com/vectordotdev/vector/issues/11995) was already merged in a new version to update my vector node and just saw that version `0.23+` does not run on CentOS 7, which I'm still using.

Building from source on a CentOS 7 machine works as a workaround?

**Comment by jszwedko** [comment]
Hi @leandrojmp !

Unfortunately we haven't been able to fix this quite yet. However, yes, building on a CentOS 7 machine should work to create a binary linked against the older version of libc.

**Comment by abh** [comment]
If this doesn't have a milestone anymore, maybe at least update the documentation to say that CentOS 7 isn't supported?  https://vector.dev/docs/setup/installation/operating-systems/centos/

**Comment by flyingflo** [comment]
Looks like cross-rs got a special target `x86_64-unknown-linux-gnu:centos` for CentOS7 with glibc 2.17. That could solve this issue, if I understand that correctly!? 
https://github.com/cross-rs/cross/commit/80891e7c216f98196a48b38fd34db6b3518f99bd

**Comment by apollo13** [comment]
I have added PR that addresses this issue. I see @bruceg assigned this to himself, I hope we didn't do the work in parallel now :)

**Comment by bruceg** [comment]
> I have added PR that addresses this issue. I see @bruceg assigned this to himself, I hope we didn't do the work in parallel now :)

Thanks for taking this up @apollo13. I had taken a look at it before @flyingflo pointed out that `cross` can do at least some of the work for us, and then moved back to other work, so no, we're not duplicating work here.

**Comment by aaronjheng** [comment]
Can we get a patch release for this?

**Comment by jszwedko** [comment]
> Can we get a patch release for this?

0.27.0 will be going out in a week and a half so I think we'll likely just loop it into that release.
